### PR TITLE
11 fault polygon region

### PR DESCRIPTION
**Pull request by chrisbc**
- various change in upstream classes for reuse / sub-classing in NSHM
 - merging Kevins work via upstream/master branch

### CODE REVIEW

File: src/scratch/UCERF3/griddedSeismicity/GridReader.java [hunk 1]
```diff
@@ -134,15 +134,33 @@ public Double apply(String s) {
 	}
 
 	private static class FnDblToKey implements Function<Double, Integer> {
+		double offset = 0.0;
```
Comment (voj) [inline src/scratch/UCERF3/griddedSeismicity/GridReader.java:137]: looks like this offset is not required

File: src/scratch/UCERF3/inversion/InversionTargetMFDs.java [hunk 2]
```diff
@@ -146,10 +146,16 @@ public InversionTargetMFDs() {
 	}
 	
 	
+	/**
```
Comment (voj) [inline src/scratch/UCERF3/inversion/InversionTargetMFDs.java]: this introduces a double-up that is addressed in merge-artifact commit. can be squashed.

File: src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java [hunk 3]
```diff
@@ -73,6 +71,11 @@ public GriddedRegion getRegion() {
 		return region;
 	}
 	
+	private FaultPolyMgr setRegion(GriddedRegion region) {
```
Comment (voj) [inline src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java]: Why are you using a private setter method? Wouldn't a constructor make more sense?

File: src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java [hunk 4]
```diff
@@ -178,28 +181,43 @@ public Region getPoly(int sectIdx) {
...
 		if (log) System.out.println("Building poly mgr...");
 		if (log) System.out.println("   section polygons");
 		mgr.polys = SectionPolygons.create(fspd, buf, null);
 		mgr.init();
 		return mgr;
 	}
 	
+	/**
+	 * Create a fault polygon manager from a list of {@code FaultSectionPrefData}.
+	 * This method assumes that  of which references the polygon of it's parent fault.
```
Comment (voj) [inline src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java:193]: I can't parse this sentence.

File: src/scratch/UCERF3/griddedSeismicity/GriddedSeisUtils.java [hunk 5]
```diff
@@ -20,18 +17,31 @@
...
+	 * Create an instance of this utility class with a region
+	 *
+	 * @param fltSectList
+	 * @param pdf pass in result of PDF allows us to use alternate SpatiolSeisPDf provider
+	 * @param buf distance in km to buffer fault polygons
+	 * @param region the user defined region
+	 */
+	public GriddedSeisUtils(List<? extends FaultSection> fltSectList, 
+			double[] pdf, double buf, GriddedRegion region) {
+		this.pdf = pdf.clone();
```
Comment (voj) [inline src/scratch/UCERF3/griddedSeismicity/GriddedSeisUtils.java]: on line 43, it's `getPDF()` instead of `clone()`. Should that be the same?

Comment (voj) [reply src/scratch/UCERF3/griddedSeismicity/GriddedSeisUtils.java]: I would also recommend to swap the order of the two lines to make it look more like the other constructor

File: src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java [hunk 6]
```diff
@@ -58,9 +58,12 @@
 	private Map<Integer, Double> nodeExtents;
 	private Map<Integer, Double> sectExtents;
 
-	private static GriddedRegion region = new CaliforniaRegions.RELM_TESTING_GRIDDED();
+	private static GriddedRegion region;
```
Comment (voj) [inline src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java]: should be merged with commit that makes this non-static

File: src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java [hunk 7]
```diff
@@ -58,9 +58,12 @@
...
+	//= new CaliforniaRegions.RELM_TESTING_GRIDDED();
```
Comment (voj) [inline src/scratch/UCERF3/griddedSeismicity/FaultPolyMgr.java]: we don't need this comment

File: src/org/opensha/sha/earthquake/faultSysSolution/ruptures/plausibility/ScalarValuePlausibiltyFilter.java [hunk 8]
```diff
@@ -161,13 +161,13 @@ public default String getRangeStr() {
 		Preconditions.checkState(acceptableRange.hasLowerBound() || acceptableRange.hasUpperBound());
```
Comment (voj) [inline src/org/opensha/sha/earthquake/faultSysSolution/ruptures/plausibility/ScalarValuePlausibiltyFilter.java:161]: use better commit message

File: src/scratch/UCERF3/inversion/InversionFaultSystemRupSet.java [hunk 9]
```diff
@@ -178,11 +178,12 @@ public InversionFaultSystemRupSet(
 			List<List<Integer>> clusterRups,
 			List<List<Integer>> clusterSects) {
 		super(branch.getValue(SlipAlongRuptureModels.class));
+		init(rupSet); //CBC this needs to happen before the following helper methods
```
Comment (voj) [inline src/scratch/UCERF3/inversion/InversionFaultSystemRupSet.java]: do we need this comment?

**Review by voj — CHANGES_REQUESTED** [review]
A few remarks and questions. Also, we should clean up the git history before creating a pull request

File: src/scratch/UCERF3/griddedSeismicity/GridReader.java (hunk 1 above)
Comment (chrisbc) [reply src/scratch/UCERF3/griddedSeismicity/GridReader.java:137]: Yes - removed it.

File: src/scratch/UCERF3/inversion/InversionFaultSystemRupSet.java (hunk 9 above)
Comment (chrisbc) [reply src/scratch/UCERF3/inversion/InversionFaultSystemRupSet.java]: Agreed we're leaving this in for follow up on root cause..

**Pull request merged**